core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java

```diff
@@ -397,21 +397,25 @@ private Blueprint determineBlueprint(StackV4Request stackRequest, Workspace work
             return null;
         }
         return executorService.submit(() -> {
-            try {
                 boolean base = blueprintService.isAmbariBlueprint(blueprint) ? shouldUseBaseAmbariImage(clusterRequest) : shouldUseBaseCMImage(clusterRequest);
                 LOGGER.info("The stack with name {} will use base image: {}", stackName, base);
                 return imageService.determineImageFromCatalog(workspace.getId(), stackRequest.getImage(), platformString, blueprint, base, user, image -> true);
-            } catch (CloudbreakImageNotFoundException | CloudbreakImageCatalogException e) {
-                throw new BadRequestException(e.getMessage(), e);
-            }
         });
     }
 
     private StatedImage getImageCatalog(Future<StatedImage> imgFromCatalogFuture) {
         return Optional.ofNullable(imgFromCatalogFuture).map(f -> {
             try {
                 return f.get(1, TimeUnit.HOURS);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            } catch (ExecutionException e) {
+                if (e.getCause() instanceof CloudbreakImageNotFoundException) {
+                    throw new BadRequestException("Image id settings are incorrect.", e);
+                }
+                if (e.getCause() instanceof CloudbreakImageCatalogException) {
+                    throw new BadRequestException("Image catalog settings are incorrect.", e);
+                }
+                throw new RuntimeException("Unkown error happened when determining image catalog", e);
+            } catch (InterruptedException | TimeoutException e) {
```

Comment:
I think we can add more specific message in these cases

Reply:
Split the to cases, also fin tuned the messages.